libbindgen/src/lib.rs

```diff
@@ -243,6 +243,22 @@ impl Builder {
         self
     }
 
+    /// Disable auto-namespacing of names if namespaces are disabled.
+    ///
+    /// By default, if namespaces are disabled, bindgen tries to mangle the
+    /// names to from `foo::bar::Baz` to look like `foo_bar_Baz`, instead of
+    /// just `Baz`.
+    ///
+    /// This option disables that behavior.
+    ///
+    /// Note that this intentionally doesn't change the names using for
+    /// whitelisting and blacklisting, that should still be mangled with the
+    /// namespaces.
```

Comment:
Should probably mention the dangers of types with the same name in different namespaces, since one might not recognize this potential footgun.